src/app/api/adventure/[adventureId]/route.ts

```diff
@@ -45,3 +46,44 @@ export async function GET(
     );
   }
 }
+
+
+export async function PATCH( req: NextRequest, { params }: { params: { adventureId: string } }){
+  const session = await getServerSession(authOptions);
+  const data = await req.json();
+  if (!session) {
+    return NextResponse.json({ error: "Unauthorized" }, { status: 401 });
+  }
+  console.log("data looks like: ", data);
+  await dbConnect();
+
+  try {
+    const adventure = await AdventureModel.findById(params.adventureId);
+    const user = await UserModel.findOne( {email: data.email})
+    console.log("user is now: ", user);
+    if (!adventure) {
+      return NextResponse.json(
+        { error: `Adventure with id ${params.adventureId} not found` },
+        { status: 404 }
+      );
+    }
+    if (!user) {
+      console.log("issue with user");
+      return NextResponse.json(
+        { error: `User with ${data.email} not found` },
+        { status: 404 }
+      );
+    }
+    
+    adventure.participants.push(user._id);
+    await adventure.save();
+    
+    return NextResponse.json({ message: "User added successfully" }, { status: 200 });
+    // Handle the request
+  } catch (error: any) {
+    return NextResponse.json(
+      { error: `Failed to add adventure user: ${error}` },
+      { status: 500 }
+    );
+  }
```

Comment:
There is going to be a conflict with the /adventure/:adventureId route with the stuff in my branch as it handles being able to edit the adventure name, description and start-end dates. 

Maybe later we can refactor this endpoint to have all of the logic, but for now I recommend making this functionality a separate endpoint?

ie. `/adventure/:adventureId/participants`. I'm thinking also to be re RESTful it should be a PUT request?